R-package/R/lgb.Dataset.R

```diff
@@ -219,7 +287,7 @@ Dataset <- R6::R6Class(
           )
 
         } else if (methods::is(private$raw_data, "dgCMatrix")) {
-          if (length(private$raw_data@p) > 2147483647L) {
+          if (length(private$raw_data@p) > .Machine$integer.max) {
```

Comment:
There might be a specific reason that this was hard-coded to this number and not to `.Machine$integer.max`, I'd have to do some investigation. Either way, if this change is not directly related to the feature "accept data frames as inputs" can you please revert this and open a separate pull request with it?

Reply:
Reverted, although it wouldn't matter because `Matrix` itself wouldn't allow to make an object that big as it is also limited by INT_MAX in the dimensions.

Reply:
thank you